src/include/duckdb/main/client_context_state.hpp

```diff
@@ -31,12 +32,13 @@ class RegisteredStateManager;
 enum class RebindQueryInfo { DO_NOT_REBIND, ATTEMPT_TO_REBIND };
 
 struct PreparedStatementCallbackInfo {
-	PreparedStatementCallbackInfo(PreparedStatementData &prepared_statement, const PendingQueryParameters &parameters)
+	PreparedStatementCallbackInfo(PreparedStatementData &prepared_statement,
+	                              const optional_ptr<case_insensitive_map_t<BoundParameterData>> &parameters)
 	    : prepared_statement(prepared_statement), parameters(parameters) {
 	}
 
 	PreparedStatementData &prepared_statement;
-	const PendingQueryParameters &parameters;
+	optional_ptr<case_insensitive_map_t<BoundParameterData>> parameters;
```

Comment:
I changed the type, because there is (I think) no way to access `PendingQueryParameters`, or `allow_stream_result`, which is necessary to create `PendingQueryParameters` in the binder. I, however, need PendingQueryParameters, because [OnExecutePrepared](https://github.com/duckdb/duckdb/pull/16757/files#diff-e3482b3527370c703bd8a1becdf712924f3f58ca45eff6fafb1e9dcd26c2f5e2R70) takes it as an argument in `info`.

Because I do not think, that it makes a difference for binding if the query is a streaming one, I omitted this parameter. If that is not OK, I'll see if there is a way to pass the information to the binder in a different way